pymc/logprob/mixture.py

```diff
@@ -483,3 +484,71 @@ def logprob_MixtureRV(
     "basic",
     "mixture",
 )
+
+
+class MeasurableIfElse(IfElse):
+    """Measurable subclass of IfElse operator."""
+
+
+MeasurableVariable.register(MeasurableIfElse)
+
+
+@node_rewriter([IfElse])
+def find_measurable_ifelse_mixture(fgraph, node):
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableIfElse):
+        return None
+
+    # Check if all components are unvalued measuarable variables
+    if_var, *base_rvs = node.inputs
```

Comment:
Would it be of any benefit to check that the `base_rvs` have matching shapes? And/or their shapes are compatible with `if_var`?

Edit: Just saw the related conversation with @lucianopaz below. I'll still leave these comments here

Reply:
`if_var` must always be a scalar. I guess you mean checking if the shape of the value variable matches the shape of the `base_rvs` from the evaluated branch. That would only make sense at evaluation time during the `logp`